vimoutliner/scripts/otl2html.py

```diff
@@ -1,216 +1,153 @@
-                    showUsage()                 # show the help
-                    sys.exit()                  # exit
-                elif (sys.argv[i] == "-p"):     # test for the slides flag
-                    slides = 1                  # set the slides flag
-                elif (sys.argv[i] == "-D"):     # test for the divisions flag
-                    div = 1                     # set the divisions flag
-                elif (sys.argv[i] == "-T"):     # test for the no-title flag
-                    showTitle = 0               # clear the show-title flag
-                elif (sys.argv[i] == "-c"):     # test for the comments flag
-                    hideComments = 1            # set the comments flag
-                elif (sys.argv[i] == "-C"):     # test for the copyright flag
-                    copyright = sys.argv[i + 1]  # get the copyright
-                    i = i + 1                   # increment the pointer
-                elif (sys.argv[i] == "-s"):     # test for the style sheet flag
-                    styleSheet = sys.argv[i + 1]  # get the style sheet name
-                    formatMode = "indent"       # set the format
-                    i = i + 1                   # increment the pointer
-                elif (sys.argv[i] == "-S"):     # test for the style sheet flag
-                    styleSheet = sys.argv[i + 1]  # get the style sheet name
-                    formatMode = "indent"       # set the format
-                    inlineStyle = 1
-                    i = i + 1                   # increment the pointer
-                elif (sys.argv[i] == "--help"):
-                    showUsage()
-                    sys.exit()
-                elif (sys.argv[i] == "-h"):
-                    showUsage()
-                    sys.exit()
-                elif (sys.argv[i] == "-H"):
-                    showSyntax()
-                    sys.exit()
-                elif (sys.argv[i][0] == "-"):
-                    print "Error!  Unknown option.  Aborting"
-                    sys.exit()
-                else:                           # get the input file name
-                    inputFile = sys.argv[i]
-
+    parser = argparse.ArgumentParser(
+        description='''
+Convert a tab-formatted outline from VIM to HTML
+
+Copyright 2001 Noel Henson All rights reserved
+
+This program accepts text outline files and converts them
+to HTML.  The outline levels are indicated by tabs. A line with no
+tabs is assumed to be part of the highest outline level.
+
+10 outline levels are supported.  These loosely correspond to the
+HTML H1 through H9 tags.  Alphabetic, numeric and bullet formats
+are also supported.
+''')
+    parser.add_argument('inputFile', nargs=1,
+                        help='Input OTL outline filename.')
+    parser.add_argument('-d', '--debug', action='store_true',
+                        help='Show debugging report.')
+    parser.add_argument('-?', '-h', '--help', action='help')
```

Comment:
This line breaks the script. See:

```
python2 otl2html.py --help
Traceback (most recent call last):
  File "otl2html.py", line 1034, in <module>
    main()
  File "otl2html.py", line 1003, in main
    conf = getArgs()
  File "otl2html.py", line 111, in getArgs
    parser.add_argument('-?', '-h', '--help', action='help')
  File "/usr/lib/python2.7/argparse.py", line 1308, in add_argument
    return self._add_action(action)
  File "/usr/lib/python2.7/argparse.py", line 1682, in _add_action
    self._optionals._add_action(action)
  File "/usr/lib/python2.7/argparse.py", line 1509, in _add_action
    action = super(_ArgumentGroup, self)._add_action(action)
  File "/usr/lib/python2.7/argparse.py", line 1322, in _add_action
    self._check_conflict(action)
  File "/usr/lib/python2.7/argparse.py", line 1460, in _check_conflict
    conflict_handler(action, confl_optionals)
  File "/usr/lib/python2.7/argparse.py", line 1467, in _handle_conflict_error
    raise ArgumentError(action, message % conflict_string)
argparse.ArgumentError: argument -h/--help: conflicting option string(s): -h, --help
```

Reply:
You could just remove the line altogether. The help options are added automatically by argparse. Yes, the `-?` would be dropped, but I think that's acceptable.